### PR TITLE
add 'namespace' and 'sharedWith' property to Mailbox. Fixes #69

### LINKED ISSUE #69: add 'namespace' and 'sharedWith' property to Mailbox to share mailboxes

### PR DESCRIPTION
**Pull request by fabienmoyon**

### CODE REVIEW

File: test/common/models/Mailbox.js [hunk 1]
```diff
@@ -144,8 +150,12 @@ describe('The Mailbox class', function() {
       var mailbox = jmap.Mailbox.fromJSONObject({}, {
         id: 'id',
         name: 'name',
+        namespace: {
+          type: 'personal'
```
Comment (fvn-linagora) [inline test/common/models/Mailbox.js]: James is using (currently) `Personal` (Pascal Case) as default value.

File: test/common/models/Mailbox.js [hunk 2]
```diff
@@ -144,8 +150,12 @@ describe('The Mailbox class', function() {
...
+        },
         parentId: 'parentId',
         role: 'inbox',
+        sharedWith: {},
```
Comment (fvn-linagora) [inline test/common/models/Mailbox.js]: No test or business logic for `shareWith` property yet ? Indeed, it involves a new data structure and some upcoming new models.

**Review by fvn-linagora — COMMENTED** [review]
a bit early or WIP, don't you think ?

**Comment by fvn-linagora** [comment]
`CHANGELOG.md` also has to be updated, I guess

**Comment by ddolcimascolo** [comment]
@fabienmoyon Can you add an entry in the changelog ?

File: test/common/models/Mailbox.js (hunk 1 above)
Comment (fabienmoyon) [reply test/common/models/Mailbox.js]: Done

File: test/common/models/Mailbox.js (hunk 2 above)
Comment (fabienmoyon) [reply test/common/models/Mailbox.js]: Exactly, we will see that after.

**Comment by fabienmoyon** [comment]
@ddolcimascolo @fvn-linagora done

**Comment by fvn-linagora** [comment]
0.0.24 has already been shipped (without updated CHANGELOG unfortunately :/ )
You should reference 0.0.25 and possibly add missing changelog for 0.0.24, based on last merged PR, I guess.

**Pull request merged**